navext/navext-runtime/src/main/java/com/foursoft/harness/navext/runtime/io/write/XmlListEmptyUnsetListener.java

```diff
@@ -0,0 +1,98 @@
+/*-
+ * ========================LICENSE_START=================================
+ * NavExt Runtime
+ * %%
+ * Copyright (C) 2019 - 2025 4Soft GmbH
+ * %%
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ * =========================LICENSE_END==================================
+ */
+package com.foursoft.harness.navext.runtime.io.write;
+
+import com.foursoft.harness.navext.runtime.JaxbModelException;
+import jakarta.xml.bind.Marshaller;
+import jakarta.xml.bind.annotation.XmlList;
+
+import java.lang.reflect.Field;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Stream;
+
+public class XmlListEmptyUnsetListener extends Marshaller.Listener {
+
+    private static final Map<Class<?>, List<Field>> XML_LIST_FIELDS_CACHE = new ConcurrentHashMap<>();
```

Comment:
I´d rather do the init in a static constructor without the concurrent hashmap. The performance-impact is minimal and it is easier to debug.